### ISSUE TITLE
ReturnToPrevious: Review responsive behaviour and a11y

### ISSUE THREAD

**Issue by eledobleefe** [grafana/grafana#82535]
In this issue, we are going to check:
1. The responsive behaviour of the RTP button group.
2. The a11y concerns regarding RTP.

**Comment by eledobleefe** [comment]
## Responsive behaviour

- The RTP is a floating button group that is centered on the bottom of the page so it keeps that placement even when the user resizes the screen.
- The minimum screen width that it supports (does not overflow) is 265px, much smaller than current mobile devices.

Example on an iPhone SE:
<img width="440" alt="Captura de pantalla 2024-02-15 a las 13 32 47" src="https://github.com/grafana/grafana/assets/65417731/87192ceb-bebe-4a37-af84-4a29ee73f729">

Example on a 265px-width screen:
<img width="449" alt="Captura de pantalla 2024-02-15 a las 13 33 45" src="https://github.com/grafana/grafana/assets/65417731/716ca537-7783-4723-949f-ea9c3131d7bc">


### **Issues found:**
1. On mobile, when the nav is open the RTP button group appears above it:
<img width="444" alt="Captura de pantalla 2024-02-15 a las 13 34 20" src="https://github.com/grafana/grafana/assets/65417731/4d8ea8e2-abd2-458f-889d-edb9798c3264">

**Comment by eledobleefe** [comment]
## Accessibility

- It passes the _WCAG Contrast checker_ analysis, both in dark and light mode:

<img alt="Captura de pantalla 2024-02-15 a las 14 06 12" src="https://github.com/grafana/grafana/assets/65417731/978801b3-67f7-49e3-98c4-47030014fddc"> 
<img  alt="Captura de pantalla 2024-02-15 a las 14 05 33" src="https://github.com/grafana/grafana/assets/65417731/8eb97452-31f0-43ef-a399-abeb382fa9f8"> 

- Is keyboard focusable. 
   - The element gets focus after navigating through all the main content. 
   - We checked using 'VoiceOver' and it reads the title 'Back to x' and the 'Close' label from the dismissable button.
   - Although it is a button group and it is indicated, using the keyboard we focus first on the main button (the one that will trigger the return) and then the closing icon.

<img width="991" alt="Captura de pantalla 2024-02-15 a las 14 03 37" src="https://github.com/grafana/grafana/assets/65417731/f2699aee-40f8-44f6-9bca-7f7e536c80fb">
<img width="992" alt="Captura de pantalla 2024-02-15 a las 14 04 33" src="https://github.com/grafana/grafana/assets/65417731/a068c267-6c2c-4b0d-bb74-41376ab8052d">

**Comment by eledobleefe** [comment]
To sum up:
1. We fixed its `zIndex` issue on https://github.com/grafana/grafana/pull/82680, so it will appear behind nav, drawers, modals and similar elements.
2. We investigated, and there are no specific a11y rules regarding button groups. We checked the following:
2.1. It is keyboard-focusable after the main page content, and it can be used through a keyboard.
2.2. The screen reader (VoiceOver) reads the labels correctly.
2.3. The colour contrast passes the WCAG Contrast checker analysis.
3. Talking about responsive behaviour, as it has a fixed width and without any changes, the RTP button group sticks centered and does not overflow even in 300px-wide screens, therefore we are not going to change anything.
